cassandra-diagnostics-reporter-datadog/src/main/java/io/smartcat/cassandra/diagnostics/reporter/DatadogReporter.java

```diff
@@ -0,0 +1,107 @@
+package io.smartcat.cassandra.diagnostics.reporter;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.timgroup.statsd.NonBlockingStatsDClient;
+import com.timgroup.statsd.StatsDClient;
+
+import io.smartcat.cassandra.diagnostics.Measurement;
+
+/**
+ * A Datadog based {@link Reporter} implementation. Query reports are reporter via Datadog HTTP API
+ */
+public class DatadogReporter extends Reporter {
+
+    private static final Logger logger = LoggerFactory.getLogger(DatadogReporter.class);
+
+    private static final String STATSD_HOST_KEY = "statsDHost";
+
+    private static final String STATSD_PORT_KEY = "statsDPort";
+
+    private static final String DEFAULT_STATSD_PORT = "8125";
+
+    private static final String KEYS_PREFIX_KEY = "keysPerfix";
+
+    private static final String DEFAULT_KEYS_PREFIX = "";
+
+    private static final String FIXED_TAGS_KEY = "fixedTags";
+
+    private static final List<String> DEFAULT_FIXED_TAGS = new ArrayList<>();
+
+    private static String hostname;
+
+    private static StatsDClient client;
+
+    /**
+     * Constructor.
+     *
+     * @param configuration Reporter configuration
+     */
+    public DatadogReporter(ReporterConfiguration configuration) {
+        super(configuration);
+
+        logger.debug("Initializing datadog client with config: {}", configuration.toString());
+
+        hostname = getHostname();
```

Comment:
Think each and every reporter will have hostname, IMO all logic to pull hostname should go to `Reporter` class and should be done same way for all reporters. All measurements should not have hostname defined in it but glued by reporter on single place.